server/dataone_register.py

```diff
@@ -144,13 +144,12 @@ def find_initial_pid(path):
     if re.search(r'^http[s]?:\/\/search.dataone.org\/#view\/', path):
         return re.sub(
             r'^http[s]?:\/\/search.dataone.org\/#view\/', '', path)
-    elif re.search(r'^http[s]?://cn.dataone.org/cn/d1/v[\d]/\w+/', path):
+    elif re.search(r'^http[s]?:\/\/cn.dataone.org\/cn(\/d1)?\/v[\d]/\w+\/', path):
+        print('sdfgsergewrgwefgewrg')
         return re.sub(
-            r'^http[s]?://cn.dataone.org/cn/d1/v[\d]/\w+/', '', path)
-    elif re.search('https:\/\/cn-stage-2.test.dataone.org\/cn\/v2\/resolve\/', path):
-        return re.sub('https:\/\/cn-stage-2.test.dataone.org\/cn\/v2\/resolve\/', path)
-    elif re.search(r'^http[s]?:\/\/cn.dataone.org\/cn\/v2\/resolve\/', path):
-        return re.sub('http[s]?:\/\/cn.dataone.org\/cn\/v2\/resolve\/', '', path)
+            r'^^http[s]?:\/\/cn.dataone.org\/cn(\/d1)?\/v[\d]/\w+\/', '', path)
```

Comment:
the double `^^` is weird to me -- is that valid or desired? The regex should also really be anchored with `\A` (start of string) and `\Z` (end of string). I may have been the one that coded this regex in the first place so take my comments with that in mind. Also there is the `d1` thing again and I see an unescaped `\` in the middle of the regex.

Maybe a more general regex would be something like

`\Ahttp[s]?:\/\/cn[a-z\-\d\.]*\.dataone\.org\/cn\/v\d\/[a-zA-Z]+\/.+\Z`

That should work for (1) http/https, (2) any cn environment, (3) any api version, (4) whether the URI has `object` or `meta` or `resolve` as the API method.